pkg/ccl/changefeedccl/sink_kafka_v2.go

```diff
@@ -328,7 +331,10 @@ func (b *kafkaBuffer) Append(ctx context.Context, key []byte, value []byte, attr
 		headers = append(headers, kgo.RecordHeader{Key: k, Value: v})
 	}
 
-	rctx := context.WithValue(ctx, mvccTSKey{}, attrs.mvcc)
+	rctx := ctx
```

Comment:
nit: technically this is still new behaviour since we didnt used to pass the ctx to the records at all. can you make it nil when disabled?